src/Uno.UI/UI/Xaml/Controls/Repeater/ElementAnimator.cs

```diff
@@ -0,0 +1,207 @@
+		{
+			if (HasBoundsChangeAnimation(element, (AnimationContext)(context), oldBounds, newBounds))
+			{
+				HasBoundsChangeAnimationsPending = true;
+				SharedContext |= context;
+				QueueElementForAnimation(new ElementInfo(
+					element,
+					AnimationTrigger.BoundsChange,
+					context,
+					oldBounds,
+					newBounds));
+			}
+		}
+
+		public bool HasShowAnimation(UIElement element, AnimationContext context)
+			=> HasShowAnimationCore(element, context);
+
+		public bool HasHideAnimation(UIElement element, AnimationContext context)
+			=> HasHideAnimationCore(element, context);
+
+		public bool HasBoundsChangeAnimation(UIElement element, AnimationContext context, Rect oldBounds, Rect newBounds)
+			=> HasBoundsChangeAnimationCore(element, context, oldBounds, newBounds);
+		#endregion
+
+		#region IElementAnimatorOverrides
+
+		protected virtual bool HasShowAnimationCore(UIElement element, AnimationContext context)
+			=> throw new NotImplementedException();
+
+		protected virtual bool HasHideAnimationCore(UIElement element, AnimationContext context)
+			=> throw new NotImplementedException();
+
+		protected virtual bool HasBoundsChangeAnimationCore(UIElement element, AnimationContext context, Rect oldBounds, Rect newBounds)
+			=> throw new NotImplementedException();
+
+		protected virtual void StartShowAnimation(UIElement element, AnimationContext context)
+			=> throw new NotImplementedException();
+
+		protected virtual void StartHideAnimation(UIElement element, AnimationContext context)
+			=> throw new NotImplementedException();
+
+		protected virtual void StartBoundsChangeAnimation(UIElement element, AnimationContext context, Rect oldBounds, Rect newBounds)
+			=> throw new NotImplementedException();
+
+		#endregion
+
+		#region IElementAnimatorProtected
+
+		protected bool HasShowAnimationsPending { get; private set; }
+
+		protected bool HasHideAnimationsPending { get; private set; }
+
+		protected bool HasBoundsChangeAnimationsPending { get; private set; }
+
+		protected AnimationContext SharedContext { get; private set; }
+
+		protected void OnShowAnimationCompleted(UIElement element)
+			=> ShowAnimationCompleted?.Invoke(this, element);
+
+		protected void OnHideAnimationCompleted(UIElement element)
+			=> HideAnimationCompleted?.Invoke(this, element);
+
+		protected void OnBoundsChangeAnimationCompleted(UIElement element)
+			=> BoundsChangeAnimationCompleted?.Invoke(this, element);
+		#endregion
+
+		private void QueueElementForAnimation(ElementInfo elementInfo)
+		{
+			m_animatingElements.Add(elementInfo);
+			if (m_animatingElements.Count == 1)
+			{
+				Windows.UI.Xaml.Media.CompositionTarget.Rendering += OnRendering;
+			}
+		}
+
+		private void OnRendering(object snd, object args)
+		{
+			Windows.UI.Xaml.Media.CompositionTarget.Rendering -= OnRendering;
+
+			using var resetState = Disposable.Create(ResetState);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'resetState'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)